homeassistant/components/zamg/sensor.py

```diff
@@ -261,17 +271,19 @@ def __init__(
     @property
     def native_value(self) -> StateType:
         """Return the state of the sensor."""
-        return self.coordinator.data[self.station_id].get(
-            self.entity_description.para_name
-        )["data"]
+        try:
+            return self.coordinator.data[self.station_id][
+                self.entity_description.para_name
+            ]["data"]
+        except (KeyError):
+            return None
 
     @property
     def extra_state_attributes(self) -> Mapping[str, str]:
         """Return the state attributes."""
-        update_time = self.coordinator.data.get("last_update", "")
+        if (update_time := self.coordinator.data["last_update"]) is not None:
+            update_time = update_time.isoformat()
         return {
-            ATTR_ATTRIBUTION: ATTRIBUTION,
-            ATTR_STATION: self.coordinator.data.get("Name"),
-            CONF_STATION_ID: self.station_id,
-            ATTR_UPDATED: update_time.isoformat(),
+            ATTR_STATION: self.coordinator.data["Name"],
+            ATTR_UPDATED: update_time or STATE_UNKNOWN,
```

Comment:
```suggestion
            ATTR_UPDATED: update_time,
```

Reply:
`update_time` could be `None` here, is it ok?

Reply:
Yes, we prefer keeping the key and set `None` as value if the value is unknown.